src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
```

Comment:
PullProofAI

Consider adding a check for the existence of `x-hub-signature-256` header before type casting to avoid potential runtime errors.
```diff
+     const signature = request.headers['x-hub-signature-256'] as string | undefined;
```



<!-- This is an auto-generated comment by PullProofAI -->